vllm/entrypoints/openai/api_server.py

```diff
@@ -116,9 +116,16 @@ async def build_async_engine_client(args) -> AsyncIterator[AsyncEngineClient]:
 
         # Build RPCClient, which conforms to AsyncEngineClient Protocol.
         async_engine_client = AsyncEngineRPCClient(port)
-        await async_engine_client.setup()
 
         try:
+            while True:
+                try:
+                    await async_engine_client.setup()
+                    break
+                except TimeoutError as e:
+                    if not rpc_server_process.is_alive():
+                        raise RuntimeError("Server crashed") from e
```

Comment:
Can you add a better error message here?

Reply:
Yes, that wasn't very informative.